design/tab_complete.md

```diff
@@ -0,0 +1,63 @@
+
+# Tab/Auto Completion #
+
+Tab completion is important for creating a CLI that provides a good user experience. This feature
+is somewhat nebulous at the moment. There are many aspects to it and the implementation will be
+dependent on what's most feasible with the command dispatch framework that we use.
+
+## Goals ##
+There are multiple different aspects of tab completion that may or may not make it into the CLI
+depending on how feasible they are to implement.
+
+*Must have:*
+* Support for bash.
+* Subcommand completion, e.g. `dcos cluster[TAB]` should give you `list attach remove rename`.
```

Comment:
missing space `dcos cluster [TAB]`

Reply:
The space wasn't left out particularly intentionally but also I don't think it needs to be there because it's perfectly valid to hit tab at that point and expect the same behavior.

Reply:
Not exactly, even though there is a single match, at this point the word to autocomplete is still `cluster`, one would have to press TAB twice in order to get the behavior you're describing.